src/coreclr/vm/interpexec.cpp

```diff
@@ -1213,6 +1370,7 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
     }
     catch (const ResumeAfterCatchException& ex)
     {
+        GCX_COOP_NO_DTOR();
```

Comment:
We should instead fix the source of the problem - remove the switch to preemptive mode from the `CallDescrWorkerUnwindFrameChainHandler`. I believe we don't need to have it there.